### ISSUE TITLE
linear eval while pretraining gradient

### ISSUE THREAD

**Issue by slala2121** [google-research/simclr#151]
1. If I understand correctly, setting linear eval to True while pretraining does not allow gradients from the supervised loss to influence the training of the encoder i.e., resnet base + projection head? That is, I should see identical optimization trajectories when only pretraining and not simultaneously running linear evaluation?

2. in particular, what happens if the input into the supervised head is a middle layer of the MLP? Does the above statement still hold?

**Comment by chentingpc** [comment]
The stop gradient operator makes sure the gradient from the label will not
be used anywhere beyond the linear classifier, so in either case, it will
be equivalent to not training the linear classifier during the pretraining.

**Comment by slala2121** [comment]
Great, thanks for the clarification.
